internal/hcp/registry/json.go

```diff
@@ -106,6 +113,11 @@ func (h *JSONRegistry) CompleteBuild(
 			name, k, pluginDetails.Description.Version,
 		)
 	}
+
+	err := h.bucket.AddMetadataToBuild(ctx, name, metadata)
```

Comment:
When the build starts we haven't registered plugins use yet, so besides the Packer version, will we have anything at this point? I'd think we'll only register metadata on completion

Reply:
I am okay with that. @JenGoldstrich What's your preference?

Reply:
Makes sense to me to only send it when we're finishing the build, I would suggest that it may be good to add metadata also for failed builds if its feasible, and not just for complete builds though.

Reply:
What's the rationale for that though? This will be replaced ultimately (or abandoned if the version isn't finished), no image has been created if the build failed, so we don't get a lot of interesting info in this case I think.

Reply:
Just noticed that this was not under `StartBuild` but under `CompleteBuild` so nevermind me anyway, this is well placed then.

Regarding debugging/troubleshooting that depends if that's a use-case we want to support/address, I'm not certain this will help in any way (maybe some convenience on the run that failed?). Not necessarily opposed to it, but I'm very uncertain of the value is all.